service/src/main/java/org/broadinstitute/listener/relay/Utils.java

```diff
@@ -54,26 +53,24 @@ public static void writeCORSHeaders(
   }
 
   public static boolean isValidOrigin(String origin, CorsSupportProperties corsSupportProperties) {
-    String stringToCheck;
+    if (origin.isEmpty() || corsSupportProperties.validHosts().contains("*")) {
+      return true;
+    }
+
     // We want to strip the protocol.
+    URL url;
     try {
-      URL url = new URL(origin);
-      stringToCheck = url.getHost();
+      url = new URL(origin);
     } catch (MalformedURLException e) {
-      stringToCheck = origin;
+      logger.error(String.format("Error parsing URL:%s. MalformedURLException: %s", origin, e.getMessage()));
+      return false;
     }
 
-    logger.info(String.format("Checking origin %s", stringToCheck));
-    logger.info(corsSupportProperties.validHosts().toString());
-    // To appeal to lambda.
-    String finalStringToCheck = stringToCheck;
-    return origin.isEmpty()
-        || corsSupportProperties.validHosts().stream()
+    return corsSupportProperties.validHosts().stream()
         .anyMatch(
             validHost ->
-                Pattern.matches(
-                    validHost.replace(".", "\\.").replace("*", ".*").replace(" ", ""),
-                    finalStringToCheck));
+                // We still need to strip out spaces!
+                validHost.trim().equals(url.getAuthority()));
```

Comment:
One last thing, and then I swear to God I'm done😅 - the `trim()` feels out of place here, the more I think about it. We could normalize the validHosts at the source, inside the CorsSupportProperties record, with an explicit canonical constructor: e.g.
```
public record CorsSupportProperties (
    String preflightMethods,
    String allowHeaders,
    String maxAge,
    String contentSecurityPolicy,
    List<String> validHosts) {
				public CorsSupportProperties (
				    String preflightMethods,
				    String allowHeaders,
    				String maxAge,
    				String contentSecurityPolicy,
    				List<String> validHosts) {
    						this.preflightMethods = preflightMethods;
    						this.allowHeaders = allowHeaders;
    						this.maxAge = maxAge;
    						this.contentSecurityPolicy = contentSecurityPolicy;
    						this.validHosts = validHosts.stream().map(host -> host.trim()).collect(Collectors.toList())
				}
}
```
I know this is 12 lines in the model to replace a single method call in the service, but it feels better to normalize this data which is used all over.

Reply:
(forgive if any of this is wrong; I'm not super familiar with streams or records, having worked mostly in older Java. But they're very cool)

Reply:
This is an excellent suggestion! Thank you. I'll make the adjustments